posts/onboarding.md

```diff
@@ -0,0 +1,525 @@
+
+> [!NOTE]
+> Some dialects, like BigQuery, treat identifiers as case-insensitive even when they're quoted, so all identifiers are normalized.
+
+Different dialects may have different normalization strategies. For instance, in Snowflake unquoted identifiers are normalized to uppercase.
+
+This example demonstrates how the identifier `Foo` is normalized to lowercase `foo` in the default sqlglot dialect and uppercase `FOO` in the Snowflake dialect:
+
+```Python
+import sqlglot
+from sqlglot.optimizer.normalize_identifiers import normalize_identifiers
+
+normalize_identifiers("Foo").sql()
+# 'foo'
+
+normalize_identifiers("Foo", dialect="snowflake").sql("snowflake")
+# 'FOO'
+```
+
+##### Qualifying tables and columns
+After normalizing the identifiers, all table and column names are qualified so there is no ambiguity about the data sources referenced by the query.
+
+This means that all table references are assigned an alias and all column names are prefixed with their source table’s name.
+
+This example demonstrates qualifying the query `SELECT col FROM tbl`, where `tbl` contains a single column `col`. Note that the table’s schema is passed as an argument to `qualify()`:
+
+```Python
+import sqlglot
+from sqlglot.optimizer.qualify import qualify
+
+schema = {"tbl": {"col": "INT"}}
+expression = sqlglot.parse_one("SELECT col FROM tbl")
+qualify(expression, schema=schema).sql()
+
+# 'SELECT "tbl"."col" AS "col" FROM "tbl" AS "tbl"'
+```
+
+The `qualify` rule also offers a set of sub-rules that further canonicalize the query.
+
+For instance, `qualify()` can expand stars such that each `SELECT *` is replaced by a projection list of the selected columns.
+
+This example modifies the previous example’s query to use `SELECT *`. `qualify()` expands the star and returns the same output as before:
+
+```Python
+import sqlglot
+from sqlglot.optimizer.qualify import qualify
+
+
+schema = {"tbl": {"col": "INT"}}
+expression = sqlglot.parse_one("SELECT * FROM tbl")
+qualify(expression, schema=schema).sql()
+
+
+# 'SELECT "tbl"."col" AS "col" FROM "tbl" AS "tbl"'
+```
+
+#### Type annotation
+Some SQL operators’ behavior changes based on the data type of its inputs.
+
+For example, in some SQL dialects `+` can be used for either adding numbers or concatenating strings. The database uses its knowledge of column data types to determine which operation should be executed in a specific situation.
+
+Like the database, SQLGlot needs to know column data types to perform some optimizations. In many cases, both transpilation and optimization need type information to work properly.
+
+Type annotation begins with user-provided information about column data types for at least one table. SQLGlot then traverses the AST, propagating that type information and attempting to infer the type returned by each AST expression.
+
+Users may need to provide column type information for multiple tables to successfully annotate all expressions in the AST.
+
+
+
+
```

Comment:
Nit: let's try to be consistent about the whitespace we use. My personal preference is to always separate sections with a single blank line. Open to alternatives as long as we stick to one style.
```suggestion
```